source/common/network/socket_impl.h

```diff
@@ -17,6 +17,11 @@ class SocketImpl : public virtual Socket {
     local_address_ = local_address;
   }
 
+  SocketPtr duplicate() override {
+    // Implementing the functionality here for all sockets is tricky because it leads
+    // into object slicing issues.
+    RELEASE_ASSERT(false, "Socket duplicate function is only for listener sockets.");
```

Comment:
I would use NOT_IMPLEMENTED here.

Reply:
Added in 8b5dda0f38661892609144ed39df64dec80293cb